server/src/main/java/io/crate/analyze/relations/DocTableRelation.java

```diff
@@ -56,20 +57,26 @@ public DocTableRelation(DocTableInfo tableInfo) {
 
     @Nullable
     @Override
-    public Reference getField(ColumnIdent path) {
-        return getField(path, Operation.READ);
+    public Reference getField(ColumnIdent path) throws UnsupportedOperationException, ColumnUnknownException {
+        Symbol ref = getField(path, Operation.READ, true);
+        assert ref instanceof Reference : "unless errorOnUnknownObjectKey is false, ref should be Reference type";
+        return (Reference) ref;
     }
 
     @Override
-    public Reference getField(ColumnIdent column, Operation operation) throws UnsupportedOperationException, ColumnUnknownException {
+    public Symbol getField(ColumnIdent column,
+                           Operation operation,
+                           boolean errorOnUnknownObjectKey) throws UnsupportedOperationException, ColumnUnknownException {
         if (operation == Operation.UPDATE) {
             ensureColumnCanBeUpdated(column);
         }
         Reference reference = tableInfo.getReadReference(column);
         if (reference == null) {
             reference = tableInfo.indexColumn(column);
             if (reference == null) {
-                return tableInfo.getDynamic(column, operation == Operation.INSERT || operation == Operation.UPDATE);
+                reference = tableInfo.getDynamic(column,
+                                                 operation == Operation.INSERT || operation == Operation.UPDATE,
+                                                 errorOnUnknownObjectKey);
```

Comment:
Why the changes here? (Other than adding the argument). If `getField` keeps returning a `Reference` the assertion and cast is not necessary.

Reply:
sorry forgot to take out, this was added for my first implementation where `getDynamic` would return Literal.NULL or DynamicReference.